### DISCUSSION TITLE
Cheap Python-Capable Single Board Computer Suggestion?

### DISCUSSION THREAD

**Discussion by silveroranges** [markqvist/Reticulum#89] [Ideas]
I have been trying to find rpi3's and rpi4's for a while, and have only found price gougers selling them which I refuse to buy from. 

Does anybody have a single board computer alternative that is cheap, and would run the reticulum network stack? 

Cheap being preferably under $40.

I have so many LoRa transmitters I just need to get the boards to use them with.

Honestly just thinking about putting a back order for 30 or 40 CM4 compute modules that are backordered until at least 03/23, since I don't see this getting better any time soon. 

Another idea I had was to find a cheap android phone and use something like Linux Deploy to see if I could get it working on that ecosystem, since you can find a ton of used android phones for around $10 on eBay. But there is a lot of unknowns with that route as well, mainly getting pyserial to recognize external hardware and if it would even work. I've got an older android phone I plan on testing once I get some extra time.

**Comment by kc1awv** [comment]
When you say 'python capable' and 'single board computer', could you clarify?

Using the common definition of single board computer, any SBC should be capable of running Python and RNS, Raspberry Pi or otherwise.

You don't need to use an SBC, you could use an old laptop, desktop, server, what have you. I've seen off-lease desktops going for pennies on ebay that are significantly more useful than an SBC.

**Comment by zenith777777** [comment]
Thin clients such as the Dell Wyse series thin clients  sell for $5 - $10 used on eBay and can often be bought in bulk for even less. They have the same relative performance and in some cases greater performance than the Raspberry Pi and it's clones. The drawbacks are they don't have the same form factor and have greater power consumption. One of the benefits of the Pi, especially the Zero model is the lower power consumption which allows for completely portable operations. 

There is also the Orange Pi which is currently available on sites like Aliexpress but it's always a gamble buying these since there is very low quality control from what I have seen.

**Reply by silveroranges** [reply]
Thanks for the replies, I forgot I had posted here until I got the email just now. I will look into the Dell thin clients, that seems like an awesome price for what I am hoping to do, but at the cost of the form factor like you said. It could be a great cheap option for somebody that values economy over size. I am really looking for something I can buy in bulk in the tens or hundreds, to create a node I can produce in scale* easily. I have coworkers interested in the reticulum network, they all volunteered to host nodes, and if I can get them all connected I could cover an area just over 80 square miles, depending on the topography/antenna options available to them.

If/when the chip shortage clears up I would love to use some compute modules. My goal is to keep the cost(my cost) of the node under $60. I don't know enough about circuitry or PCB design to design something from scratch unfortunately. I got some Libre SBC's to check out, I will also check out the Orange Pi as well. 

At this point I am just window shopping to get some ideas, my end goal being a producer of reticulum nodes that are adaptable**, with 100% of the profit off the nodes going to Mark to keep the project growing. 

Just finished summer classes, and heading into fall now though so time is short :(

*scale is a few hundred/month, or less depending on demand.
**one idea I had would be to sell a basic node with LoRa, but have accessories that are plug and play eg;  plug into a 2 way radio jack to turn it into a modem, depending upon FCC allowing it. I'm thinking accessories could be big, car kits, even fire/burglar alarm systems that are connected no matter what.

**Comment by markqvist** [comment]
The whole raspi shortage is pretty annoying, since it was a platform that was very easy to deploy to, and quite reliable with good quality control, and overall low power consumption. This has obviously changed now that they are so hard to get.

As @kc1awv and @zenith777777 has suggested, almost any computer that runs some sort of mainline Linux will work just fine. Even many devices that have custom/patched/old kernels, and many devices running embedded Linux works just fine too. I have Reticulum and nomadnet running on a couple of ebook readers ;)

Unfortunately I can't give any definite suggestions on boards that are easily available in bulk and cheap enough for mass deployment, other than what has already been suggested. I am struggling just getting enough devices for development and testing.

My current focus in trying to alleviate this is to make sure that Reticulum can run on almost anything you might have available, and to move as fast as I can towards having the stack running directly on microcontrollers like the ESP32 as well. Such low-powered devices will not be efficient for running big transport instances with lots of traffic (for that an SBC or any old computer will be more suitable), but it will allow you to let small devices handle all the application node tasks themselves.

This is really the most useful approach I think, and will open up so many interesting possibilities. I am working as hard as I can towards this, but it is a non-trivial problem field to get right, and it needs to be done right to work well :)

Also, thank you very much for your support and donations, you have no idea how much it helps and means. Funding is quite scarce for me at the moment, so every bit really helps out a lot. Thank you so much.
